source/m2mbase.cpp

```diff
@@ -188,20 +179,16 @@ void M2MBase::set_under_observation(bool observed,
             _report_handler->set_under_observation(observed);
         }
     } else {
-        if(_report_handler) {
-            delete _report_handler;
-            _report_handler = NULL;
-        }
+        delete _report_handler;
+        _report_handler = NULL;
     }
 }
 
 void M2MBase::set_observation_token(const uint8_t *token, const uint8_t length)
 {
-    if(_token) {
-         free(_token);
-         _token = NULL;
-         _token_length = 0;
-    }
+     free(_token);
```

Comment:
This is potential crash. _token will be NULL for the first time and this API call will crash.